src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/LaunchProfiles/LaunchSettingsQueryVersionProviderExport.cs

```diff
@@ -0,0 +1,218 @@
+        /// Used to keep track of whether or not a new subscriber has received its initial snapshot.
+        /// </remarks>
+        private long _processedNotificationCount;
+        /// <remarks>
+        /// The current set of versions.
+        /// </remarks>
+        private QueryDataVersions _versions = QueryDataVersions.Empty;
+        /// <remarks>
+        /// The most recent set of versions sent to subscribers.
+        /// </remarks>
+        private QueryDataVersions _latestPostedVersions = QueryDataVersions.Empty;
+
+        [ImportingConstructor]
+        public LaunchSettingsQueryVersionProviderExport(LaunchSettingsQueryVersionProvider provider)
+        {
+            _provider = provider;
+#pragma warning disable RS0030 // Do not used banned APIs
+            _processingBlock = DataflowBlockSlim.CreateActionBlock<Action>(callback => callback(), nameFormat: nameof(LaunchSettingsQueryVersionProviderExport));
+#pragma warning restore RS0030 // Do not used banned APIs
+
+            _provider.BindToVersionPublisher(this);
+        }
+
+        public void UpdateVersions(ImmutableDictionary<string, long> versions)
+        {
+            lock (SyncObject)
+            {
+                _versions = QueryDataVersions.Empty.AddRange(versions);
+
+                PostVersionUpdate();
+            }
+        }
+
+        public IDisposable SubscribeChangeNotifications(IObserver<QueryDataSourceChangeNotification> observer)
+        {
+            EnsureInitialized();
+
+            lock (SyncObject)
+            {
+                if (_observers.Contains(observer))
+                {
+                    Requires.Fail($"This observer is already subscribed to change notifications from the {nameof(LaunchSettingsQueryVersionProviderExport)}.");
+                }
+
+                _observers = _observers.Add(observer);
+
+                // We don't immediately send the current set of versions to the observer. Instead,
+                // we post a message to send that data later. However, by that point the observer
+                // may have already received an up-to-date set of versions. To avoid double-sending
+                // the data we keep track of how many version updates we've already sent. If that
+                // number hasn't changed when we process the posted message, we know the observer
+                // has not received any version data yet.
+                long processedNotificationCountAtTimeOfSubscription = _processedNotificationCount;
+
+                _processingBlock.Post(() =>
+                {
+                    bool observerRequiresNotification = false;
+                    lock (SyncObject)
+                    {
+                        if (_observers.Contains(observer)
+                            && processedNotificationCountAtTimeOfSubscription == _processedNotificationCount)
+                        {
+                            observerRequiresNotification = true;
+                        }
```

Comment:
Instead of an if-statement and then assigning `true`, couldn't you just assign it the predicate directly?
```suggestion
                        observerRequiresNotification = _observers.Contains(observer) && processedNotificationCountAtTimeOfSubscription == _processedNotificationCount);
```